lib/topological_inventory/amazon/collector.rb

```diff
@@ -31,8 +31,16 @@ def initialize(source, access_key_id, secret_access_key, batch_size: 1_000, poll
       end
 
       def collect!
-        entity_types.each do |entity_type|
-          process_entity(entity_type)
+        loop do
+          begin
```

Comment:
Hm don't think you need the begin, rescue at the same level as the loop will work and save some indentation

Reply:
the is for ruby 2.5, so it won't pass the CI

Reply:
:+1: